packages/manager/modules/web-paas/src/components/modify-plan/modify-plan.controller.js

```diff
@@ -0,0 +1,278 @@
+  }
+
+  $onInit() {
+    this.isAdding = false;
+    this.stepperIndex = 0;
+    this.isEditingTemplate = false;
+    this.isEditingOffers = false;
+    this.isGettingAddons = false;
+    this.isGettingCheckoutInfo = false;
+    this.prices = null;
+    this.additionalLicenceEdit = false;
+
+    this.project = {
+      region: null,
+      offer: null,
+      name: null,
+      template: {
+        createNew: true,
+        templateUrl: null,
+      },
+    };
+    if (this.selectedProject) {
+      this.setStepperIndex();
+      this.project.name = this.selectedProject.projectName;
+      this.onPlanSelect(this.selectedProject);
+    }
+  }
+
+  setStepperIndex() {
+    this.stepperIndex = 1;
+  }
+
+  refreshMessages() {
+    this.messages = this.messageHandler.getMessages();
+  }
+
+  onPlanSubmit() {
+    if (this.isChangingPlan()) {
+      if (this.shouldRemoveExtraLicences()) {
+        this.loadOptions();
+      }
+      return this.upgradePlan();
+    }
+    return this.loadCapabilities(this.project.offer);
+  }
+
+  onPlanSelect(product) {
+    this.selectedPlan = product.selectedPlan;
+    this.project.offer = product.selectedPlan.planCode;
+  }
+
+  onTemplateSelect(templateUrl) {
+    this.project.template.templateUrl = templateUrl;
+  }
+
+  onTemplateFocus() {
+    this.isEditingTemplate = true;
+  }
+
+  onTemplateSubmit() {
+    this.loadOptions();
+    this.isEditingTemplate = false;
+  }
+
+  onOfferFocus() {
+    this.isEditingOffers = true;
+  }
+
+  onOfferSubmit() {
+    this.isEditingOffers = false;
+  }
+
+  getConfiguration() {
+    const config = [
+      {
+        label: 'region',
+        value: this.project.region,
+      },
+      {
+        label: 'project_title',
+        value: this.project.name,
+      },
+    ];
+    if (!this.project.template.createNew) {
+      config.push({
+        label: 'options_url',
+        value: this.project.template.templateUrl,
+      });
+    }
+    return config;
+  }
+
+  loadCapabilities(planCode) {
+    this.loadingCapabilities = true;
+    return this.WebPaas.getCapabilities(planCode)
+      .then((capabilities) => {
+        this.capabilities = capabilities;
+        if (
+          this.capabilities.regions &&
+          this.capabilities.regions.length === 1
+        ) {
+          [this.project.region] = this.capabilities.regions;
+        }
+      })
+      .catch(() =>
+        this.Alerter.alertFromSWS(
+          this.$translate.instant('web_paas_add_project_region_na'),
+          'error',
+          this.alerts.add,
+        ),
+      )
+      .finally(() => {
+        this.loadingCapabilities = false;
+      });
+  }
+
+  scrollToTop() {
+    this.$timeout(() => {
+      document.getElementById('web-pass-add-header').scrollIntoView({
+        behavior: 'smooth',
+      });
+      document.getElementById('web-pass-add-alert').focus();
+    });
+  }
+
+  onPlatformOrderSuccess(checkout) {
+    if (checkout && checkout.prices && checkout.prices.withTax.value > 0) {
```

Comment:
Shouldn't we use the `product-offer` component here ?

Reply:
Tried using `<product-offer>` component but we have multiple addons here and its not yet supported by the component.  Also this component is used for 3 different workflows. So decided to not use the component.